rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -296,36 +296,37 @@ MasterGraph::build()
     start_processing();
     return Status::OK;
 }
-Image *
-MasterGraph::create_loader_output_image(const ImageInfo &info)
-{
+
+rocalTensor *
+MasterGraph::create_loader_output_tensor(const rocalTensorInfo &info) {
     /*
-    *   NOTE: Output image for a source node needs to be created as a regular (non-virtual) image
+    *   NOTE: Output tensor for a source node needs to be created as a regular (non-virtual) tensor
     */
-    auto output = new Image(info);
-
+    auto output = new rocalTensor(info);
     if(output->create_from_handle(_context) != 0)
-        THROW("Creating output image for loader failed");
+        THROW("Creating output tensor for loader failed");
 
-    _internal_images.push_back(output);
+    _internal_tensors.push_back(output);
 
     return output;
 }
 
-Image *
-MasterGraph::create_image(const ImageInfo &info, bool is_output)
-{
-    auto* output = new Image(info);
-    // if the image is not an output image, the image creation is deferred and later it'll be created as a virtual image
-    if(is_output)
-    {
-        if (output->create_from_handle(_context) != 0)
-            THROW("Cannot create the image from handle")
+rocalTensor *
+MasterGraph::create_tensor(const rocalTensorInfo &info, bool is_output) {
+    auto *new_tensor = new rocalTensor(info);
```

Comment:
please use output instead of new_tensor

Reply:
Renamed 
new_tensor -> output
output -> output_ext (since this tensor will be returned and accessed by the user)